src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -415,6 +438,13 @@ def get_command_table(module_name=None):
         logger.debug("Loaded all modules in %.3f seconds. "
                      "(note: there's always an overhead with the first module loaded)",
                      cumulative_elapsed_time)
+    try:
+        # We always load extensions even if the appropriate module has been loaded
+        # as an extension could override the commands already loaded.
```

Comment:
Private/Public previews need the ability to modify existing commands so that was the thinking behind this.

From a security perspective, as is, it's up to the user to verify that the extension their about to install is from a 'safe' source. If we have an 'Extension Service' of sorts then there would be a marketplace that can be trusted but that's not in this PR (something for future?).